source/world_builder/features/continental_plate_models/composition/random.cc

```diff
@@ -0,0 +1,138 @@
+
+          prm.declare_entry("bound values", Types::Array(Types::Double(1), (0, 1)),
+                            "Bounds within which we want to generate a random compositional value corresponding to the compositional field. "
+                            "The left and the right value determines the lower and upper bound of the compositional field, respectively.");
+
+          prm.declare_entry("operation", Types::String("replace", std::vector<std::string> {"replace", "replace defined only", "add", "subtract"}),
+                            "Whether the value should replace any value previously defined at this location (replace) or "
+                            "add the value to the previously define value. Replacing implies that all compositions not "
+                            "explicitly defined are set to zero. To only replace the defined compositions use the replace only defined option.");
+
+        }
+
+        void
+        Random::parse_entries(Parameters &prm, const std::vector<Point<2>> &coordinates)
+        {
+          min_depth_surface = Objects::Surface(prm.get("min depth",coordinates));
+          min_depth = min_depth_surface.minimum;
+          max_depth_surface = Objects::Surface(prm.get("max depth",coordinates));
+          max_depth = max_depth_surface.maximum;
+
+          compositions = prm.get_vector<unsigned int>("compositions");
+          bound_values = prm.get_vector<double>("bound values");
+          operation = string_operations_to_enum(prm.get<std::string>("operation"));
+        }
+
+
+        double
+        Random::get_composition(const Point<3> & /*position_in_cartesian_coordinates*/,
+                                const Objects::NaturalCoordinate &position_in_natural_coordinates,
+                                const double depth,
+                                const unsigned int composition_number,
+                                double composition,
+                                const double  /*feature_min_depth*/,
+                                const double  /*feature_max_depth*/) const
+        {
+          if (depth <= max_depth && depth >= min_depth)
+            {
+              const double min_depth_local = min_depth_surface.constant_value ? min_depth : min_depth_surface.local_value(position_in_natural_coordinates.get_surface_point()).interpolated_value;
+              const double max_depth_local = max_depth_surface.constant_value ? max_depth : max_depth_surface.local_value(position_in_natural_coordinates.get_surface_point()).interpolated_value;
+              if (depth <= max_depth_local &&  depth >= min_depth_local)
+                {
+                  for (unsigned int i =0; i < compositions.size(); ++i)
+                    {
+                      if (compositions[i] == composition_number)
+                        {
+                          const double composition_ = ((double)rand()) / RAND_MAX * (bound_values[1] - bound_values[0]) + bound_values[0];
```

Comment:
This works, but the following would be more in line with how the rest of the code generates random numbers.

```suggestion
                          std::uniform_real_distribution<> dist(min_value,max_value);
                          const double composition_ = dist(world->get_random_number_engine());
```